monai/transforms/transforms.py

```diff
@@ -749,28 +767,40 @@ def __call__(self, img):
         return adjuster(img)
 
 
-class PadImageEnd(Transform):
-    """Performs padding by appending to the end of the data all on one side for each dimension.
+class SpatialPad(Transform):
+    """Performs padding to the data, symmetric for all sides or all on one side for each dimension.
      Uses np.pad so in practice, a mode needs to be provided. See numpy.lib.arraypad.pad
      for additional details.
 
     Args:
-        out_size (list): the size of region of interest at the end of the operation.
-        mode (string): a portion from numpy.lib.arraypad.pad is copied below.
+        spatial_size (list): the spatial size of output data after padding.
+        method (str): pad image symmetric on every side or only pad at the end sides. default is 'symmetric'.
+        mode (str): one of the following string values or a user supplied function: {'constant', 'edge', 'linear_ramp',
+            'maximum', 'mean', 'median', 'minimum', 'reflect', 'symmetric', 'wrap', 'empty', <function>}
+            for more details, please check: https://docs.scipy.org/doc/numpy/reference/generated/numpy.pad.html
     """
 
-    def __init__(self, out_size, mode):
-        assert isinstance(out_size, (list, tuple)), 'out_size must be list or tuple.'
-        self.out_size = out_size
+    def __init__(self, spatial_size, method='symmetric', mode='constant'):
+        assert isinstance(spatial_size, (list, tuple)), 'spatial_out_size must be list or tuple.'
+        self.spatial_size = spatial_size
+        assert method in ('symmetric', 'end'), 'unsupported padding type.'
+        self.method = method
         assert isinstance(mode, str), 'mode must be str.'
         self.mode = mode
 
     def _determine_data_pad_width(self, data_shape):
-        return [(0, max(self.out_size[i] - data_shape[i], 0)) for i in range(len(self.out_size))]
+        if self.method == 'symmetric':
+            pad_width = list()
+            for i in range(len(self.spatial_size)):
+                width = max(self.spatial_size[i] - data_shape[i], 0)
+                pad_width.append((width // 2, width - (width // 2)))
+            return pad_width
+        else:
+            return [(0, max(self.spatial_size[i] - data_shape[i], 0)) for i in range(len(self.spatial_size))]
```

Comment:
now I feel this padding isn't very useful compared with the symmetric padding

Reply:
Hi @wyli ,

Thanks for your review, the end padding is used by sliding window inference.
Thanks.

Reply:
is it because it's easier to implement the sliding window? symmetric padding is more appropriate for that as well

Reply:
anyway perhaps we could have another iteration in the future to fix it if needed

Reply:
OK, sounds good.